gui/gui/src/TGFileBrowser.cxx

```diff
@@ -1243,14 +1243,9 @@ void TGFileBrowser::DoubleClicked(TGListTreeItem *item, Int_t /*btn*/)
          if (clname) {
             TClass *cl = TClass::GetClass(clname);
             TString name = ((TKey *)obj)->GetName();
-            // check in memory first
-            void *add = gDirectory->FindObject((char *) name.Data());
-            // if the object was not found read it (with its cycle number) from the file
-            if (!add) {
-               name += ";";
-               name += ((TKey *)obj)->GetCycle();
-               add = gDirectory->FindObjectAny((char *) name.Data());
-            }
+            name += ";";
+            name += ((TKey *)obj)->GetCycle();
```

Comment:
If you have the key and want to always recreate the object, what not use ```TKey::ReadObj``` ?

Reply:
Thanks Philippe, `gDirectory->FindObjectAny` always recreate it as well, but I think I will keep it the way it is, since I prefer to always read the object, since it can change in the file...

Reply:
But essentially the code above and FindObjectAny:
* Build an "id" for the key
* Search for that id and find it back the same key (maybe)
* Call TKey::ReadObj and the key.
So at best it is a waste of cpu ( ;) )

Reply:
I think I will just let the code the way it is...